transmogrifier/sources/transformer.py

```diff
@@ -153,233 +208,308 @@ def parse_source_file(cls, source_file: str) -> Iterator[JSON | Tag]:
         pass
 
     @abstractmethod
-    def get_optional_fields(self, source_record: JSON | Tag) -> Optional[dict]:
+    def transform(self, source_record: JSON | Tag) -> Optional[TimdexRecord]:
         """
-        Retrieve optional TIMDEX fields from a source record.
+        Transform a source record into a TIMDEX record.
 
-        Must be overridden by source subclasses.
+        Must be overridden by format subclasses.
 
         Args:
             source_record: A single source record.
         """
-        return {}
+        pass
 
-    @classmethod
     @abstractmethod
-    def get_main_titles(cls, source_record: JSON | Tag) -> list[Tag | str]:
+    def get_required_fields(self, source_record: JSON | Tag) -> dict:
         """
-        Retrieve main title(s) from an source record.
+        Get required TIMDEX fields from a source record.
 
-        Must be overridden by source subclasses.
+        Must be overridden by format subclasses.
 
         Args:
             source_record: A single source record.
         """
-        return []
+        pass
 
     @classmethod
-    @abstractmethod
-    def get_source_record_id(cls, source_record: JSON | Tag) -> str:
+    def get_main_titles(cls, source_record: JSON | Tag) -> list[str]:
```

Comment:
Should this also be an `@abstractmethod`, requiring it's either defined by `XMLTransformer`, `JSONTransformer`, or their children assuming those two don't define it?

Reply:
Good catch!